src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -2228,6 +2228,26 @@ public static Map<String,String> getInvertiblePinMap(Series series, Unisim unisi
 		return invertPinMap;
 	}
 
+	/**
+	 * Copy a cell from a design as a new design.
+	 * @param src The source design with or without implementation
+	 * @param cellName The full hierarchy cell name to be extracted as another design
+	 * @param  keepStaticRouting A flag to indicate whether to keep the routes of static nets or not.
+	 * @return A newly created design copied from the source
+	 */
+	public static Design createDesignFromCell(Design src, String cellName, boolean keepStaticRouting) {
+		EDIFNetlist srcCellNetlist = EDIFTools.createNewNetlist(src.getNetlist().getHierCellInstFromName(cellName).getInst());
+		EDIFTools.ensureCorrectPartInEDIF(srcCellNetlist, src.getPartName());
+		Design d2 = new Design(srcCellNetlist);
+		d2.setAutoIOBuffers(false);
+		d2.setDesignOutOfContext(true);
+
+		// TODO: Skip this step if the design was not implemented.
+		Map<String, String> cellMap = Collections.singletonMap(cellName, "");
+		DesignTools.copyImplementation(src, d2,  keepStaticRouting, true, true, true, cellMap);
+		return d2;
+	}
+
```

Comment:
Is this a necessary part of this PR?

Reply:
This could be moved to PR 390.  The only reason I see for the move is it call copyImplementation with 3 flags as true, which I don't know if it is generic.

If we don't think, the flags set are not generic, we should move it to PR 390.

Reply:
Yes, I think that's a good idea since we aren't using nor testing it in this PR.

Reply:
move to PR 390